fearless/Modules/Staking/SelectValidatorsFlow/ValidatorInfo/Interactor/ValidatorInfoInteractorBase.swift

```diff
@@ -0,0 +1,32 @@
+import RobinHood
+
+class ValidatorInfoInteractorBase: ValidatorInfoInteractorInputProtocol {
+    weak var presenter: ValidatorInfoInteractorOutputProtocol!
+
+    internal let singleValueProviderFactory: SingleValueProviderFactoryProtocol
+    private let assetId: WalletAssetId
+
+    private var priceProvider: AnySingleValueProvider<PriceData>?
+
+    init(
+        singleValueProviderFactory: SingleValueProviderFactoryProtocol,
+        walletAssetId: WalletAssetId
+    ) {
+        self.singleValueProviderFactory = singleValueProviderFactory
+        assetId = walletAssetId
+    }
+
+    func setup() {
+        priceProvider = subscribeToPriceProvider(for: assetId)
+    }
+}
+
+extension ValidatorInfoInteractorBase: SingleValueSubscriptionHandler {
+    func handlePrice(result: Result<PriceData?, Error>, for _: WalletAssetId) {
+        presenter.didReceivePriceData(result: result)
+    }
+}
+
+extension ValidatorInfoInteractorBase: SingleValueProviderSubscriber {}
+
+extension ValidatorInfoInteractorBase: AnyProviderAutoCleaning {}
```

Comment:
`clear(dataProvider:` is not called, so I guess `AnyProviderAutoCleaning` is redundant